### ISSUE TITLE
GitHubPollRequestPoller:  pr['base']['ref']  string indices must be integers

### ISSUE THREAD

**Issue by tardyp** [buildbot/buildbot#3252]
if owner and repo are misconfigured, then a strange error is given:

```
          File "/usr/lib64/python2.7/site-packages/twisted/internet/defer.py", line 1301, in _inlineCallbacks
            result = g.send(result)
          File "/usr/lib64/python2.7/site-packages/buildbot/changes/github.py", line 207, in _processChanges
            base_branch = pr['base']['ref']
        exceptions.TypeError: string indices must be integers
```

would need to check the error of rest api, instead of sending the json directly for processing.

**Comment by aliceinwire** [comment]
solved by using the repo name in the GitHubPollRequestPoller repo argument

**Comment by noc0lour** [comment]
Yep, That's the error you usually get if you exceede the rate limiting. Wanting to fix that for quite some time now.

**Comment by tardyp** [comment]
> solved by using the repo name in the GitHubPollRequestPoller repo argument

Indeed, but the error should be more obvious.

**Comment by noc0lour** [comment]
@tardyp you can assign me and I'll take a closer look the next couple of days into handing the error return things in the PR poller.

**Comment by s0undt3ch** [comment]
This also happens if the repo is private and the user accessing it has no permissions. GitHub sends a 404 like the repo didn't even exist which is not handled appropriately.
